ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter.js

```diff
@@ -0,0 +1,370 @@
+    ${({ type, theme }) =>
+      type === 'DATE'
+        ? `min-width: calc(19 * ${theme.minimumFontSize}) !important;`
+        : null}
+    ${({ type }) => (type === 'EMPTY' ? 'display: none !important;' : null)}
+    ${({ type, theme }) =>
+      type === 'LOCATION' || type === 'GEOMETRY'
+        ? `margin-top: ${
+            theme.minimumSpacing
+          } !important; display: block !important;`
+        : null}
+  `
+
+const Filter = withListenTo(
+  class Filter extends React.Component {
+    filterAttribute
+    filterInput
+    constructor(props) {
+      super(props)
+      const comparator = props.model.get('comparator') || 'CONTAINS'
+      let attribute = props.model.get('type') || 'anyText'
+      if (
+        props.includedAttributes &&
+        !props.includedAttributes.includes(attribute)
+      ) {
+        attribute = props.includedAttributes[0]
+      }
+
+      this.state = {
+        comparator,
+        attribute,
+        editing: props.editing,
+        type: metacardDefinitions.metacardTypes[attribute].type,
+        filteredAttrs: this.getFilteredAttributeList(),
+      }
+      props.model.set('type', attribute)
+      props.model.set('comparator', comparator)
+    }
+
+    componentDidMount = () => {
+      this.props.listenTo(
+        this.props.model,
+        'change:comparator',
+        this.updateComparator
+      )
+    }
+
+    componentWillReceiveProps = ({ editing }) => {
+      this.setState({ editing })
+    }
+
+    render() {
+      const filterDropdownModel = new DropdownModel({
+        value: this.state.comparator,
+      })
+      const filterComparator = new FilterComparatorDropdownView({
+        model: filterDropdownModel,
+        modelForComponent: this.props.model,
+      })
+      this.filterAttribute = DropdownView.createSimpleDropdown({
+        list: this.state.filteredAttrs,
+        defaultSelection: [this.state.attribute],
+        hasFiltering: true,
+      })
+
+      this.props.listenTo(
+        this.filterAttribute.model,
+        'change:value',
+        this.updateAttribute
+      )
+
+      this.determineInput()
+      if (this.state.editing) {
+        this.filterAttribute.turnOnEditing()
+        filterComparator.turnOnEditing()
+        const property =
+          this.filterInput.model instanceof ValueModel
+            ? this.filterInput.model.get('property')
+            : this.filterInput.model
+        property.set('isEditing', true)
+      } else {
+        this.filterAttribute.turnOffEditing()
+        filterComparator.turnOffEditing()
+        const property =
+          this.filterInput.model instanceof ValueModel
+            ? this.filterInput.model.get('property')
+            : this.filterInput.model
+        property.set(
+          'isEditing',
+          this.props.isForm === true || this.props.isFormBuilder === true
+        )
+      }
+      return (
+        <React.Fragment>
+          <div className="filter-rearrange">
```

Comment:
Could the css in this render method be moved into styled components?

Reply:
Is there a way to reference custom elements in styled components? For example on this line: https://github.com/codice/ddf/blob/91c444e541b5041e8d6d29018984363bf6e85d92/ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.less#L42.
I don't see any examples of us doing that in our codebase, but I could be looking in the wrong places

Reply:
You would want to get rid of class selectors altogether and have a styled component for each element that needs styling